src/Symfony/Component/Mime/Part/DataPart.php

```diff
@@ -28,17 +28,28 @@ class DataPart extends TextPart
     private $filename;
     private $mediaType;
     private $cid;
-    private $handle;
 
-    /**
-     * @param resource|string $body
-     */
     public function __construct($body, string $filename = null, string $contentType = null, string $encoding = null)
     {
         unset($this->_parent);
 
+        $path = null;
+        if (\is_string($body) && str_starts_with($body, 'file://')) {
```

Comment:
can't this convention lead to trivial leaks?
we'd better pass a value object to pass a file, that's how curl does it with https://php.net/CURLFile

Reply:
Done @nicolas-grekas